### PR TITLE
Report Stream Lab Order Sender Caching

### PR DESCRIPTION
**Pull request by jorg3lopez**
# Report-Stream Lab Order Sender Caching

This PR build on top of the Caching class. It refactors the existing caching logic in reportStreamLabOrderSender, by using the caching class.

## Issue
#148

### CODE REVIEW

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java [hunk 1]
```diff
@@ -147,13 +149,13 @@ protected String requestToken() throws UnableToSendLabOrderException {
...
                 "report-stream-sender-private-key-" + ApplicationContext.getEnvironment();
-        String key = getCachedPrivateKey();
+        String key = this.keyCache.get(senderPrivateKey);
         if (key != null) {
             return key;
         }
 
         key = secrets.getKey(senderPrivateKey);
-        setCachedPrivateKey(key);
+        this.keyCache.put(senderPrivateKey, key);
```
Comment (halprin) [inline app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java:158]: With this change, you can now delete the `getCachedPrivateKey` method, `setCachedPrivateKey` method, and the `cachedPrivateKey` field in this class.

**Review by halprin — CHANGES_REQUESTED** [review]
Looks like you've got some conflicts too.

File: app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java (hunk 1 above)
Comment (jorg3lopez) [reply app/src/main/java/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSender.java:158]: Great find.

**Review by halprin — APPROVED** [review]
Nice!

**Pull request merged**